### PR TITLE
[Process] Turn getIterator() args to flags & add ITER_SKIP_OUT/ERR modes

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #8454 |
| License | MIT |
| Doc PR | - |

Targeted at 3.1

### CODE REVIEW

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -503,41 +508,49 @@ public function getIncrementalOutput()
...
-     * @param bool $clearOutput Whether to clear or keep output in memory.
+     * @param int $flags A bit field of Process::ITER_* flags.
      *
      * @throws LogicException in case the output has been disabled
      * @throws LogicException In case the process is not started
      *
      * @return \Generator
      */
-    public function getIterator($blocking = true, $clearOutput = true)
+    public function getIterator($flags = 0)
```
Comment (romainneutron) [inline src/Symfony/Component/Process/Process.php]: One of the best advantage of this is to be able to pipe a process in another process.
We could use the same behavior as unix `|` and forward only stdout (use `self:: ITER_SKIP_ERR)` by default).

WDYT?

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: :+1: good catch, I added a test case chaining two processes, works like a charm!

File: src/Symfony/Component/Process/Tests/ProcessTest.php [hunk 2]
```diff
@@ -1342,6 +1342,21 @@ public function testNonBlockingNorClearingIteratorOutput()
         $this->assertSame($expectedOutput, $output);
     }
 
+    public function testChainedProcesses()
+    {
+        $p1 = new Process(self::$phpBin.' -r '.escapeshellarg('fwrite(STDERR, 123); fwrite(STDOUT, 456);'));
```
Comment (stof) [inline src/Symfony/Component/Process/Tests/ProcessTest.php]: shouldn't this use ``getProcess`` ?

**Comment by stof** [comment]
:+1: for this (and should be merged in 3.1 to avoid having to maintain BC for the boolean-based signature as it is a new method in 3.1 anyway)

Status: reviewed

**Comment by romainneutron** [comment]
👍

File: src/Symfony/Component/Process/Tests/ProcessTest.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Tests/ProcessTest.php]: We can hav only one getProces per test case. And there is not need to configure differently the process for sigchild. So no need.

Comment (stof) [inline src/Symfony/Component/Process/Tests/ProcessTest.php]: Was not aware of ``getProcess`` being usable only once per test. so fine with me

**Pull request merged**